### PR TITLE
switchlink_address unit test

### PR DESCRIPTION
**Pull request by aashishkuma**
This PR contains the UT written for switchlink_address.cc belonging to krnlmon/switchlink directory.
Test cases added:
1. addIpv4Address
2. deleteIpv4Address
3. addIpv6Address
4. deleteIpv6Address

### CODE REVIEW

**Comment by aashishkuma** [comment]
Test Cases Execution:
Output:
----------------------------------------------------------
Running main() from /home/akumar/dev_dir/open-source/ipdk.recipe/setup/gtest/googletest/src/gtest_main.cc
[==========] Running 4 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 4 tests from SwitchlinkAddressTest
[ RUN      ] SwitchlinkAddressTest.addIpv4Address
[       OK ] SwitchlinkAddressTest.addIpv4Address (0 ms)
[ RUN      ] SwitchlinkAddressTest.deleteIpv4Address
[       OK ] SwitchlinkAddressTest.deleteIpv4Address (0 ms)
[ RUN      ] SwitchlinkAddressTest.addIpv6Address
[       OK ] SwitchlinkAddressTest.addIpv6Address (0 ms)
[ RUN      ] SwitchlinkAddressTest.deleteIpv6Address
[       OK ] SwitchlinkAddressTest.deleteIpv6Address (0 ms)
[----------] 4 tests from SwitchlinkAddressTest (0 ms total)
[----------] Global test environment tear-down
[==========] 4 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 4 tests.
<end of output>
Test time =   0.00 sec

Test Passed.
"switchlink_address_test" end time: May 31 16:29 IST
"switchlink_address_test" time elapsed: 00:00:00

**Comment by aashishkuma** [comment]
image

File: switchlink/switchlink_address_test.cc [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+ */
+
+TEST_F(SwitchlinkAddressTest, addIpv4Address) {
+  struct ifaddrmsg hdr = {
+      .ifa_family = AF_INET,
+      .ifa_prefixlen = 24,
+      .ifa_index = 1,
+  };
+
+  int prefix_len = 0;
```
Comment (ffoulkes) [inline switchlink/switchlink_address_test.cc:154]: I'd recommend defining `prefix_len` closer to the point at which it's used. See below.

**Review by ffoulkes — APPROVED** [review]
Good work!

I have no problems with the test itself. All of my comments relate to readability.

I keep harping on readability because the more unit tests I read (and write), the more aware I become of just how bloody hard they are to understand. The people who say "tests are good documentation" don't know what they're talking about.

Anything we can do to reduce the cognitive load (the amount of thinking you have to do in order to understand the code) will make the test easier to review and easier to maintain. We don't test the tests, so we depend on inspection (reviewing the code) to ensure its quality.

Are there any negative cases you can test? (This might boost your test coverage...)

**Comment by aashishkuma** [comment]
image

**Comment by aashishkuma** [comment]
> Are there any negative cases you can test? (This might boost your test coverage...)
Added 2 negative test cases as you pointed out. Shared the updated coverage report as well. 

Just to add, to cover each and every possible scenario a function could return, the dummy function will become equally intelligent in terms of the logic we are going to put in it. For complex functions, dummy functions will become equally complex.
What are your thoughts?

**Review by ffoulkes — APPROVED** [review]
LGTM

**Comment by ffoulkes** [comment]
> Just to add, to cover each and every possible scenario a function could return, the dummy function will become equally intelligent in terms of the logic we are going to put in it. For complex functions, dummy functions will become equally complex. What are your thoughts?

Not worth the added complexity for the few extra percent it would eke out.

**Review by nupuruttarwar — APPROVED** [review]
LGTM

**Pull request merged**